quickjs.h

```diff
@@ -665,9 +670,9 @@ JS_EXTERN JSValue JS_GetPropertyStr(JSContext *ctx, JSValue this_obj,
 JS_EXTERN JSValue JS_GetPropertyUint32(JSContext *ctx, JSValue this_obj,
                                        uint32_t idx);
 
-int JS_SetPropertyInternal(JSContext *ctx, JSValue this_obj,
-                           JSAtom prop, JSValue val,
-                           int flags);
+JS_EXTERN int JS_SetPropertyInternal(JSContext *ctx, JSValue this_obj,
```

Comment:
Why expose this, if it's labeled internal?

Reply:
What is the meaning of this `JS_EXTERN`  tag?  My understanding is that these symbols are made visible for programs using a dynamic link version of the library.  `JS_SetPropertyInternal` is used by the `JS_SetProperty` inline function, hence must be a public symbol if we compile quickjs as a dynamic library.
Note that since the library is essentially compiled from a single source file, we avoid the classic issue regarding *visible* and *hidden* global symbols.
Whether `JS_SetProperty` should be implemented inline or just be an entry point is another question that is worth investigating.  A number of functions, macros and types currently exposed in **quickjs.h** should probably be made private to the implementation.

Reply:
Again, unrelated to the JS_NewNumber work, so I'd rather have it as a separate PR.

That said, I see all the other JS_SetPropertyXXX are small functions in the C file which end up calling JS_SetPropertyInternal with JS_PROP_THROW, so I think the best course here would be to drop JS_SetPropertyInternal altogether from the header file and implement JS_SetProperty in the C file.